File: C-sharp/Labwork 5/MainFlow/Capturer/DateTimeCapturer.cs
namespace Labwork_5.MainFlow.Capturer
{
    public class DateTimeCapturer
    {
        public static TimeOnly CaptureTime()
        {
            TimeOnly time = new TimeOnly();
            bool valueIsValid = false;

            while (!valueIsValid)
            {
                valueIsValid = true;

                try
                {
                    System.Console.Write("Please, enter the time: ");
                    time = TimeOnly.Parse(Console.ReadLine());
                }
                catch (FormatException ex)
                {
                    System.Console.WriteLine(ex.Message);
                    valueIsValid = false;
                }
            }

            return time;
        }

        public static DateOnly CaptureDate()
        {
            DateOnly date = new DateOnly();
            bool valueIsValid = false;

            while (!valueIsValid)
            {
                valueIsValid = true;

                try
                {
                    System.Console.Write("Please, enter the date: ");
                    date = DateOnly.Parse(Console.ReadLine());
                }
                catch (FormatException ex)
                {
                    System.Console.WriteLine(ex.Message);
                    valueIsValid = false;
                }
            }

            return date;
        }
    }
}

File: Python/Labwork 5/Capturer/datetime_capturer.py
from datetime import *

class DateTimeCapturer:
    @staticmethod
    def capture_time():
        result = time()
        value_is_valid = False

        while not value_is_valid:
            value_is_valid = True

            try:
                result = time.fromisoformat(input("Please, enter the time: "))
            except ValueError as ve:
                print(ve)
                value_is_valid = False

        return result

    @staticmethod
    def capture_date():
        result = date(1, 1, 1)
        value_is_valid = False

        while not value_is_valid:
            value_is_valid = True

            try:
                iso_date = input("Please, enter the date: ")
                result = date.fromisoformat(iso_date)
            except ValueError as ve:
                print(ve)
                value_is_valid = False
            except TypeError as te:
                print(te)
                value_is_valid = False

        return result